File: src/main/java/com/wikia/tibia/repositories/CorpseRepository.java
package com.wikia.tibia.repositories;

import com.wikia.tibia.gateways.CorpseGateway;
import com.wikia.tibia.objects.Corpse;

public class CorpseRepository extends WikiObjectRepository {

    public CorpseRepository() {
        super(Corpse.class, new CorpseGateway());
    }
}

File: src/main/kotlin/com/wikia/tibia/repositories/CorpseRepository.kt
package com.wikia.tibia.repositories

import com.wikia.tibia.enums.Contract
import com.wikia.tibia.gateways.WikiObjectGateway
import com.wikia.tibia.objects.Corpse

class CorpseRepository : WikiObjectRepository<Corpse>(
  wikiObjectClass = Corpse::class.java,
  wikiObjectGateway = WikiObjectGateway(Contract.CORPSES)
)
